File: Java/1859_Sorting_the_Sentence.java
class Solution {
    public String sortSentence(String s) {
        String[] words = s.split(" ");
        StringBuilder sb = new StringBuilder();
        
        Map<Integer, String> map = new TreeMap<>();
        for(String word : words){
            int index = word.charAt(word.length()-1) - '0';
            map.put(index, word.substring(0, word.length()-1));
        }
        
        for(Map.Entry<Integer, String> entry : map.entrySet()) {
            Integer key = entry.getKey();
            String value = entry.getValue();
            
            sb.append(value + " ");
        }
        return sb.deleteCharAt(sb.length()-1).toString();
    }
}

File: CS/1859_Sorting_the_Sentence.cs
public class Solution {
    public string SortSentence(string s) {
        string[] sArray = s.Split(" ");
        int n = sArray.Length;
        string[] rArray = new string[n];
        for(int i = 0; i < n; i++){
            string item = sArray[i];
            int k = item.Length;
            //Console.WriteLine(item[k-1]='1');
            rArray[item[k-1] - '1'] = item.Substring(0, k-1);
        }
        return String.Join(" ", rArray);
        /*
        string result = rArray[0];
        for(int i = 1; i < n; i++){
            result += " " + rArray[i];
        }
        return result;
        */
    }
}
